code/__HELPERS/mobs.dm

```diff
@@ -111,3 +140,375 @@ proc/add_logs(mob/user, mob/target, what_done, var/admin=1, var/object=null, var
 		target.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been [what_done] by [user ? "[user.name][(ismob(user) && user.ckey) ? "([user.ckey])" : ""]" : "NON-EXISTANT SUBJECT"][object ? " with [object]" : " "][addition][newhealthtxt]</font>")
 	if(admin)
 		log_attack("[user ? "[user.name][(ismob(user) && user.ckey) ? "([user.ckey])" : ""]" : "NON-EXISTANT SUBJECT"] [what_done] [target ? "[target.name][(ismob(target) && target.ckey)? "([target.ckey])" : ""]" : "NON-EXISTANT SUBJECT"][object ? " with [object]" : " "][addition][newhealthtxt]")
+
+//Mutant name generation for each race. If it gets too big, we might to start using config files like normal names do.
+
+/proc/name_lizard()
+	var/generated_name
+	var/attempts = 10
+	for(var/i=1, i<=attempts, i++)
+		//first word, should be a verb
+		generated_name = pick("Runs",
+								"Hisses",
+								"Chases",
+								"Hunts",
+								"Wags",
+								"Hugs",
+								"Brews",
+								"Scratches",
+								"Slithers",
+								"Plays,"
+								"Awakes",
+								"Burns",
+								"Rolls",
+								"lifts")
+		generated_name += "-"
+		//second word
+		generated_name += pick("My",
+								"His",
```

Comment:
No idea where that would get the right gender.
Their name isn't suposed to refence themselves anyway. "Hisses-Your-Sun" for instance to refer to a 3rd person.
A male called "Wags-Her-Tail" is entirely possible, simply means it's a lizard skirt-chaser.
Gotta change "Runs" to "Runs With" though.